examples/models/llama/model.py

```diff
@@ -244,33 +244,24 @@ def __init__(self, **kwargs):
             )
 
         missing, unexpected = None, None
-        try:
-            # assign=True: load params/buffers by assignment instead of performing an in-place copy.
-            # Because we are using device="meta", tensors do not have memory associated with them
-            # and an in-place copy is a no-op. Use assign=True in load_state_dict for this scenario.
-
-            # Also, the checkpoint is loaded and dtype promoted to the transformer's dtype, which is
-            # by default initialized to fp32. This is fine because every other supported type
-            # losslessly converts to fp32, so we don't lose precision here.
-            if checkpoint:
-                missing, unexpected = self.model_.load_state_dict(
-                    checkpoint,
-                    strict=False,
-                    assign=True,
-                )  # self.model_ = Transformer(gptconf)
-            else:
-                print("Checkpoint not provided, defaulting weights to zeros.")
-                self.model_.to_empty(device="cpu")
-                for p in self.model_.parameters():
-                    p.data.fill_(0)
-                for b in self.model_.buffers():
-                    b.data.fill_(0)
-        except RuntimeError as e:
-            print(
-                f"Could not load checkpoint into mode and will defaulting weights to zeros due to error: {e}."
-            )
-            # Need to provide concrete (empty) values for meta-initialized tensors for quantization.
+        # assign=True: load params/buffers by assignment instead of performing an in-place copy.
+        # Because we are using device="meta", tensors do not have memory associated with them
+        # and an in-place copy is a no-op. Use assign=True in load_state_dict for this scenario.
+
+        # Also, the checkpoint is loaded and dtype promoted to the transformer's dtype, which is
+        # by default initialized to fp32. This is fine because every other supported type
+        # losslessly converts to fp32, so we don't lose precision here.
+        if checkpoint:
```

Comment:
yea i like this. If the user gives a checkpoint that doesn't work, we should fail, and not hide it under the rug and give them uninitialized model